### PR TITLE
[APP-4420] Add portal client chart

### PR DESCRIPTION
**Pull request by fpopa**
*Add a label `automerge` to automatically merge the PR after approval and passed checks.*

## Description

Already applied this in testing2, it works:
image

```
│ datafold-backoffice-679dcfcbf4-xllpw [2024-11-19 22:10:12,372][PID:1][INFO][datafold.cli.ws_client] Connected to server: wss://portal.datafold.io/backoffice/v1/ws                        │
```

I ended up using `terraform apply -replace="helm_release.datafold"` because I could not get terraform to pick up the chart changes when I was debugging.
Very interested in understanding how we're rolling out the helm chart version upgrade on all our environments.

### CODE REVIEW

File: charts/datafold/charts/backoffice/values.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+podAnnotations: {}
+podLabels: {}
+
+podSecurityContext:
+  runAsUser: 1000
+  runAsNonRoot: true
+
+securityContext:
+  allowPrivilegeEscalation: true
```
Comment (gtoonstra) [inline charts/datafold/charts/backoffice/values.yaml:32]: false

File: charts/datafold/charts/portal-client/values.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+replicaCount: 1
+image:
+  repository: gcr.io/datadiff-mm/datafold-server-onprem
```
Comment (gtoonstra) [inline charts/datafold/charts/portal-client/values.yaml:3]: The gcr would be really old and deprecated on google. Are we sure we don't have a us.docker.dev type of url for this?  GCR is practically gone.

Comment (fpopa) [reply charts/datafold/charts/portal-client/values.yaml:3]: Oh, I see now - will test it out.

I noticed we have the same thing on df-shell, but interestingly, I had the new code available in the portal-client pod that I deployed via this image.

however I don't see the [image here](https://console.cloud.google.com/artifacts/docker/datadiff-mm/us/datafold?hl=en&inv=1&invt=AbiHDw&project=datadiff-mm), so I don't know where it comes from
image

Comment (fpopa) [reply charts/datafold/charts/portal-client/values.yaml]: image
image

These were already used, as this image.repository value wasn't used when computing the image path:
>           image: "{{ .Values.global.shellRepository }}/server:{{ .Values.global.datafoldVersion }}"

File: charts/datafold/charts/portal-client/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+{{- end }}
+
+{{/*
+Datadog annotations
+*/}}
+{{- define "portal-client.datadog.annotations" -}}
+{{- if (eq .Values.global.datadog.install true) }}
+ad.datadoghq.com/{{ .Chart.Name }}.logs: >-
+  [{
+    "source": "datafold-server-onprem",
```
Comment (gtoonstra) [inline charts/datafold/charts/portal-client/templates/_helpers.tpl:72]: Let's change the source here, or everything gets mixed up.

File: charts/datafold/charts/portal-client/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    "service": "datafold-server-onprem",
```
Comment (gtoonstra) [inline charts/datafold/charts/portal-client/templates/_helpers.tpl:73]: And service

**Review by gtoonstra — APPROVED** [review]
Just update the datadog labels

**Comment by gtoonstra** [comment]
ALso let's update the PR title.

**Pull request merged**